invenio_app_ils/cli.py

```diff
@@ -876,8 +876,27 @@ def add_random_relations(relation_type):
                 documents, randint(2, min(5, len(documents)))
             )
 
-            objs.append(random_docs[0])
-            for record in random_docs[1:]:
+            for record in random_docs:
+                while True:
+                    doc = random.choice(documents)
+                    if doc["pid"] != record["pid"] and (
```

Comment:
it would be easier with (pseudo code)
```
try:
    rr.add(.....
    .....
except RelationError:
    handle exception accordingly
```

you wouldn't have to check all the cases in the if statement, because they are already checked in the relations module, if they are not fulfilled, the exception is raised.